radix-engine/src/system/system_callback.rs

```diff
@@ -113,42 +150,110 @@ pub struct System<C: SystemCallbackObject> {
 
     fn init<S: BootStore>(
         store: &S,
         executable: &Executable,
-        execution_config: &ExecutionConfig,
-        init_input: C::InitInput,
+        init_input: SystemInit<C::InitInput>,
     ) -> Result<Self, BootloadingError> {
-        let costing_parameters = {
+        let mut system_parameters = {
             let system_boot = store
                 .read_substate(
                     TRANSACTION_TRACKER.as_node_id(),
                     BOOT_LOADER_PARTITION,
                     &SubstateKey::Field(BOOT_LOADER_SYSTEM_SUBSTATE_FIELD_KEY),
                 )
                 .map(|v| scrypto_decode(v.as_slice()).unwrap())
-                .unwrap_or(SystemBoot::V1 {
-                    costing_parameters: CostingParameters::default(),
-                });
+                .unwrap_or(SystemBoot::V1(SystemParameters::default()));
 
             match system_boot {
-                SystemBoot::V1 { costing_parameters } => costing_parameters,
+                SystemBoot::V1(system_parameters) => system_parameters,
             }
         };
-        let callback = C::init(store, init_input)?;
 
-        let mut modules = SystemModuleMixer::new(
-            execution_config.enabled_modules,
-            execution_config.network_definition.clone(),
+        let callback = C::init(store, init_input.callback_init)?;
+
+        let enabled_modules = {
+            let mut enabled_modules = EnabledModules::AUTH | EnabledModules::TRANSACTION_RUNTIME;
+            if !executable.is_system() {
+                enabled_modules |= EnabledModules::LIMITS;
+                enabled_modules |= EnabledModules::COSTING;
+            };
+
+            if init_input.enable_kernel_trace {
+                enabled_modules |= EnabledModules::KERNEL_TRACE;
+            }
+            if init_input.execution_trace.is_some() {
+                enabled_modules |= EnabledModules::EXECUTION_TRACE;
+            }
+
+            if let Some(system_overrides) = &init_input.system_overrides {
+                if system_overrides.disable_auth {
+                    enabled_modules &= !EnabledModules::AUTH;
+                }
+                if system_overrides.disable_costing {
+                    enabled_modules &= !EnabledModules::COSTING;
+                }
+                if system_overrides.disable_limits {
+                    enabled_modules &= !EnabledModules::LIMITS;
+                }
+            }
+
+            enabled_modules
+        };
+
+        if let Some(system_overrides) = init_input.system_overrides {
+            if let Some(costing_override) = system_overrides.costing_parameters {
+                system_parameters.costing_parameters = costing_override;
+            }
+
+            if let Some(limits_override) = system_overrides.limit_parameters {
+                system_parameters.limit_parameters = limits_override;
+            }
+
+            if let Some(network_definition) = system_overrides.network_definition {
```

Comment:
Yeah, this was a tricky one as we need to be able to pass in a network definition pre-bottlenose and we can't just use a general default since we have test networks. Open to suggestions here though.

Reply:
The node does not need to pass in a network override in perpetuity just during pre-bottlenose (and mainnet does not need to pass this in at all since mainnet is default).

I think the current method of override is still simplest.

Reply:
> The node does not need to pass in a network override in perpetuity just during pre-bottlenose (and mainnet does not need to pass this in at all since mainnet is default).
>
> I think the current method of override is still simplest.

In practice though, we want the executor to be stateless / have no knowledge about what protocol version it's running -- so it will always needs to provide the network override.

Therefore the network written to the substate store is effectively never used. 

Perhaps instead we could make this not-an-override, and if they are both provided, we assert that the provided network id is equal to the network id that we load from the substate?

Reply:
> In practice though, we want the executor to be stateless / have no knowledge about what protocol version it's running -- so it will always needs to provide the network to the engine.

The executor is stateless for mainnet since mainnet is the default. If testnets could also be restarted then it would also be stateless. I think the simplicity of using overrides for all system configuration outweighs the awkwardness of using override for network definition.

Further I think the ability to override network definition could be useful if trying to simulate execution on another network.